pkg/controller/jobframework/webhook/defaulter.go

```diff
@@ -58,3 +70,27 @@ func (h *losslessDefaulter) Handle(ctx context.Context, req admission.Request) a
 	}
 	return response
 }
+
+func (h *losslessDefaulter) fieldErrors(raw []byte) ([]json.FieldError, error) {
+	obj := h.object.DeepCopyObject()
+	strictErrors, err := json.UnmarshalStrict(raw, obj, json.DisallowUnknownFields)
+	if err != nil {
+		return nil, err
+	}
+	fieldErrors := make([]json.FieldError, len(strictErrors))
+	for i, err := range strictErrors {
+		fieldErrors[i] = err.(json.FieldError)
+	}
+	return fieldErrors, nil
+}
+
+func checkIsUnknownField(patchOperation jsonpatch.JsonPatchOperation, fieldErrors []json.FieldError) bool {
+	for _, fieldError := range fieldErrors {
+		fieldPath := fieldError.FieldPath()
+		fieldPath = strings.ReplaceAll(fieldPath, ".", "/")
```

Comment:
It's likely we need to chang thing like `a.b[c].d` to `a.b.c.d`

Reply:
Yeah, I still thinking how to optimize it.

Reply:
Done